common/src/main/cpp/extensions/openxr_fb_body_tracking_extension_wrapper.cpp

```diff
@@ -0,0 +1,359 @@
+	{ XRBodyTracker::JOINT_RIGHT_PINKY_FINGER_PHALANX_PROXIMAL, XR_BODY_JOINT_RIGHT_HAND_LITTLE_PROXIMAL_FB, Quaternion(0.5, 0.5, -0.5, 0.5) },
+	{ XRBodyTracker::JOINT_RIGHT_PINKY_FINGER_PHALANX_INTERMEDIATE, XR_BODY_JOINT_RIGHT_HAND_LITTLE_INTERMEDIATE_FB, Quaternion(0.5, 0.5, -0.5, 0.5) },
+	{ XRBodyTracker::JOINT_RIGHT_PINKY_FINGER_PHALANX_DISTAL, XR_BODY_JOINT_RIGHT_HAND_LITTLE_DISTAL_FB, Quaternion(0.5, 0.5, -0.5, 0.5) },
+	{ XRBodyTracker::JOINT_RIGHT_PINKY_FINGER_TIP, XR_BODY_JOINT_RIGHT_HAND_LITTLE_TIP_FB, Quaternion(0.5, 0.5, -0.5, 0.5) },
+};
+
+OpenXRFbBodyTrackingExtensionWrapper *OpenXRFbBodyTrackingExtensionWrapper::singleton = nullptr;
+
+OpenXRFbBodyTrackingExtensionWrapper *OpenXRFbBodyTrackingExtensionWrapper::get_singleton() {
+	if (singleton == nullptr) {
+		singleton = memnew(OpenXRFbBodyTrackingExtensionWrapper());
+	}
+	return singleton;
+}
+
+OpenXRFbBodyTrackingExtensionWrapper::OpenXRFbBodyTrackingExtensionWrapper() :
+		OpenXRExtensionWrapperExtension() {
+	ERR_FAIL_COND_MSG(singleton != nullptr, "An OpenXRFbBodyTrackingExtensionWrapper singleton already exists.");
+
+	request_extensions[XR_FB_BODY_TRACKING_EXTENSION_NAME] = &fb_body_tracking_ext;
+
+	singleton = this;
+}
+
+OpenXRFbBodyTrackingExtensionWrapper::~OpenXRFbBodyTrackingExtensionWrapper() {
+	cleanup();
+}
+
+void OpenXRFbBodyTrackingExtensionWrapper::_bind_methods() {
+}
+
+void OpenXRFbBodyTrackingExtensionWrapper::cleanup() {
+	fb_body_tracking_ext = false;
+}
+
+uint64_t OpenXRFbBodyTrackingExtensionWrapper::_set_system_properties_and_get_next_pointer(void *next_pointer) {
+	system_body_tracking_properties.type = XR_TYPE_SYSTEM_BODY_TRACKING_PROPERTIES_FB;
+	system_body_tracking_properties.next = next_pointer;
+	system_body_tracking_properties.supportsBodyTracking = false;
+	return reinterpret_cast<uint64_t>(&system_body_tracking_properties);
+}
+
+godot::Dictionary OpenXRFbBodyTrackingExtensionWrapper::_get_requested_extensions() {
+	godot::Dictionary result;
+	for (auto ext : request_extensions) {
+		godot::String key = ext.first;
+		uint64_t value = reinterpret_cast<uint64_t>(ext.second);
+		result[key] = (godot::Variant)value;
+	}
+	return result;
+}
+
+void OpenXRFbBodyTrackingExtensionWrapper::_on_instance_created(uint64_t instance) {
+	if (fb_body_tracking_ext) {
+		bool result = initialize_fb_body_tracking_extension((XrInstance)instance);
+		if (!result) {
+			UtilityFunctions::print("Failed to initialize fb_body_tracking extension");
+			fb_body_tracking_ext = false;
+		}
+	}
+}
+
+void OpenXRFbBodyTrackingExtensionWrapper::_on_instance_destroyed() {
+	cleanup();
+}
+
+void OpenXRFbBodyTrackingExtensionWrapper::_on_session_created(uint64_t instance) {
+	// Skip if not enabled
+	if (!is_enabled()) {
+		return;
+	}
+
+	// Create the body-tracker handle
+	XrBodyTrackerCreateInfoFB createInfo = { XR_TYPE_BODY_TRACKER_CREATE_INFO_FB };
+	createInfo.bodyJointSet = XR_BODY_JOINT_SET_DEFAULT_FB;
```

Comment:
We're standardizing on using braced-initialization for enums initialization. See https://github.com/GodotVR/godot_openxr_vendors/pull/100#issuecomment-1981726339 for reference.

Reply:
Switched to brace initialization.